### PR TITLE
CheckStateConsistency use cached source state instead of raw DB for using the cachedNode

### PR DESCRIPTION
**Pull request by KimKyungup**
## Proposed changes

This PR changes from a state without cache (type `DBManager`) to a state with trieNodeCache (type `state.Database` ).
(Nodes are fetched from either fastcache or `srcState` and stored in `dstState`.)
This can help to read tries from old DB with trieNodeCache (fastcache).

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: blockchain/state_migration.go [hunk 1]
```diff
@@ -100,12 +100,12 @@ func (bc *BlockChain) migrateState(rootHash common.Hash) error {
 
 	start := time.Now()
 
-	srcCachedDB := bc.StateCache().TrieDB()
-	targetDB := statedb.NewDatabase(&stateTrieMigrationDB{bc.db})
+	srcState := bc.StateCache()
+	targetState := state.NewDatabase(&stateTrieMigrationDB{bc.db})
```
Comment (winnie-byun) [inline blockchain/state_migration.go:104]: The word `target` seems vague. Both old DB and new DB can be target in migration. Can you change to `dstState` like  in the test cases?
```suggestion
	dstState := state.NewDatabase(&stateTrieMigrationDB{bc.db})
```

Comment (aidan-kwon) [inline blockchain/state_migration.go:104]: Why did you change `statedb.NewDatabase` to `state.NetDatabase`?

Comment (KimKyungup) [reply blockchain/state_migration.go:104]: I changed it for `CheckStateConsistency` use a state with fast cache.

Comment (KimKyungup) [reply blockchain/state_migration.go]: reflected. :)

**Comment by KimKyungup** [comment]
Please review this commit. :)
https://github.com/klaytn/klaytn/pull/552/commits/c28fe5d9698e62beb150260c3bc3bc16116f93b9

**Comment by KimKyungup** [comment]
PTAL @winnie-byun @ehnuje 
I rebased this from Dev branch.

File: blockchain/state_migration.go [hunk 2]
```diff
@@ -103,12 +103,12 @@ func (bc *BlockChain) migrateState(rootHash common.Hash) (returnErr error) {
 
 	start := time.Now()
 
-	srcCachedDB := bc.StateCache().TrieDB()
-	targetDB := statedb.NewDatabase(&stateTrieMigrationDB{bc.db})
+	srcState := bc.StateCache()
+	dstState := state.NewDatabase(&stateTrieMigrationDB{bc.db})
```
Comment (winnie-byun) [inline blockchain/state_migration.go:107]: I am confused after the rebase. The purpose of this PR is to use fastcache for `dstState`, right? But, this line means you are not using the fastcache.
Even though `dstState` is used in `CheckStateConsistency`, it is db with no cache.

Comment (KimKyungup) [reply blockchain/state_migration.go:107]: This PR makes `CheckStateConsistency` can use a state with a cache for old DB.
For new DB, we should not set the same cache. :)

So this code is right.

Comment (winnie-byun) [reply blockchain/state_migration.go:107]: Do you mean the code below (this is from `CheckStateConsistency`) makes a new fastcache?
```
oldState, err := New(root, oldDB)
newState, err := New(root, newDB)
```
If not, can you tell me which part of the code is using a fastcache?

Comment (winnie-byun) [reply blockchain/state_migration.go:107]: Now I understand :) 
The fastcache is used for `srcState` (in code`srcState := bc.StateCache()`).
Nodes are fetched from either fastcache or `srcState` and stored in `dstState`.

File: blockchain/state/sync_test.go [hunk 3]
```diff
@@ -220,17 +220,18 @@ func TestIterativeStateSyncBatched(t *testing.T)    { testIterativeStateSync(t,
...
 	// Create a random state to copy
-	srcDb, srcRoot, srcAccounts := makeTestState(t)
+	srcState, srcRoot, srcAccounts := makeTestState(t)
 
 	// Create a destination state and sync with the scheduler
-	dstDb := database.NewMemoryDBManager()
-	sched := NewStateSync(srcRoot, dstDb, statedb.NewSyncBloom(1, dstDb.GetMemDB()), nil)
+	dstDiskDb := database.NewMemoryDBManager()
+	dstState := NewDatabase(dstDiskDb)
+	sched := NewStateSync(srcRoot, dstDiskDb, statedb.NewSyncBloom(1, dstDiskDb.GetMemDB()), nil)
```
Comment (ehnuje) [inline blockchain/state/sync_test.go:228]: Just a minor question, however, what `sched` stands for?

File: blockchain/state/iterator.go [hunk 4]
```diff
@@ -187,14 +186,14 @@ func (it *NodeIterator) retrieve() bool {
 }
 
 // CheckStateConsistency checks the consistency of all state/storage trie of given two state database.
-func CheckStateConsistency(oldDB database.DBManager, newDB database.DBManager, root common.Hash, mapSize int, quit chan struct{}) error {
+func CheckStateConsistency(oldDB Database, newDB Database, root common.Hash, mapSize int, quit chan struct{}) error {
 	// Create and iterate a state trie rooted in a sub-node
-	oldState, err := New(root, NewDatabase(oldDB))
```
Comment (ehnuje) [inline blockchain/state/iterator.go:192]: Not related to this PR, however, why do we have same `NewDatabase` function in both `state` and `statedb` packages?

Comment (ehnuje) [reply blockchain/state/iterator.go:192]: They look really confusing.. we should change their names..

Comment (KimKyungup) [reply blockchain/state/iterator.go:192]: Right. Original code has both function. We need to refine.

File: blockchain/state/sync_test.go (hunk 3 above)
Comment (KimKyungup) [reply blockchain/state/sync_test.go:228]: It is stateSync object, `sched` word is used also in `fasysync`.
